src/compiler/checker.ts

```diff
@@ -32029,6 +32029,52 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         );
     }
 
+    function areTupleLikeProperties(properties: Symbol[]): boolean {
+        return properties.length > 0 && every(properties, p => p.escapedName === "length" as __String || /^\d+$/.test(p.escapedName.toString()));

```

Comment:
And, if I'm not mistaken this won't work either:
```ts
function test(arg: { 1: (arg: number) => void; [k: number]: (...args: never) => void }) {}
test([() => {}, (arg) => {}] as const);
```

I'd try to avoid ignoring a type if it can't be proven to be incompatible. It should be easier to land on a more correct solution this way